### PR TITLE
File structure simplification

### PR DESCRIPTION
**Pull request by koresar**
I found it hard to follow the proposed file structure for the set on one-liner functions. This PR simplifies the project structure.

Also, this implements the stampit-org/stamp-specification#59 terminology.

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 import compose from './compose';
-import init from './init';
-import isStamp from './is-stamp';
-import isDescriptor from './is-descriptor';
-import isComposable from './is-composable';
+const isFunction = obj => typeof obj === 'function';
+const isObject = obj => !!obj && (typeof obj === 'function' || typeof obj === 'object');
 
+const isDescriptor = obj => isObject(obj);
```
Comment (troutowicz) [inline source/index.js]: We don't want to duck type the descriptor?

Comment (unstoppablecarl) [inline source/index.js]: I think the descriptor could be `{}` and still be valid so I'm not sure what we would check for.

**Comment by unstoppablecarl** [comment]
I think these should also be clearly defined in the stamp-specification. In the readme we should have a clear explanation of what each object is and what is tested for like the code below.

``` js
const isDescriptor = obj => isObject(obj);
const isStamp = obj => isFunction(obj) && isFunction(obj.compose) && isDescriptor(obj.compose);
const isComposable = obj => isDescriptor(obj) || isStamp(obj);
```

Looks good :+1:

File: source/index.js (hunk 1 above)
Comment (koresar) [inline source/index.js]: @troutowicz yes we don't. The properties of a descriptor can all be missing. It's a legal case. See https://github.com/stampit-org/stamp-specification/issues/59

**Comment by koresar** [comment]
There was a clarification PR about a month ago @unstoppablecarl. See https://github.com/stampit-org/stamp-specification/pull/60

If you feel it should be improved in any way - please suggest. Thanks :)

File: source/index.js (hunk 1 above)
Comment (troutowicz) [inline source/index.js]: Sounds good.

Comment (koresar) [inline source/index.js]: I'll shorten it down to

``` js
const isDescriptor = isObject;
```

:)

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "clean": "rimraf build && mkdir build",
     "lint": "eslint .",
-    "test": "babel-node test/index.js",
+    "test": "tape -r babel/register test/index.js",
```
Comment (ericelliott) [inline package.json:9]: Why do this?

Comment (koresar) [inline package.json:9]: To enable debugging. There is that tool called `iron-node`. It work in the following way - in your command line you replace `node` with `iron-node`.

When I prepend `iron-node` to `babel-node` the debugging doesn't work.

```
iron-node babel-node test/index.js
```

The received errors are quite difficult to solve.

Whereas, if I prepend `iron-node` to the `tape` command line it just works:

```
iron-node tape -r babel/register test/index.js
```

That was the only reason really. I hope that's fine.

**Comment by ericelliott** [comment]
I use a debug script with iron-node. See [this](https://github.com/keystonejs/react-easy-universal/blob/master/source/test/debug.js) [example](https://github.com/keystonejs/react-easy-universal/blob/master/package.json#L24). (Note: those are two different links. =)

**Comment by koresar** [comment]
Nice! One more file in /test directory is okay I believe. If I were you I'd add a comment to this file saying something like "// this is used only for unit test debugging with iron-node

Otherwise people might want to delete it unintentionally ☺
